src/strategies/__tests__/spidStrategy.test.ts

```diff
@@ -0,0 +1,22 @@
+import { IDP_IDS, loadFromRemote } from "../spidStrategy";
+
+describe("loadFromRemote", () => {
+  it("should reject if the fetch of IdP metadata fails", () => {
+    const notExistingUrl = "http://0.1.2.3/index.html";
+    const result = loadFromRemote(notExistingUrl, IDP_IDS);
+    return expect(result).rejects.toEqual(expect.any(Error));
+  });
+
+  it("should reject an error if the fetch of IdP metadata returns no useful data", () => {
+    const wrongIdpMetadataUrl = "http://www.example.com";
+    const result = loadFromRemote(wrongIdpMetadataUrl, IDP_IDS);
+    return expect(result).rejects.toEqual(expect.any(Error));
+  });
+
+  it("should resolve with the fetched IdP options", () => {
+    const validIdpMetadataUrl =
+      "https://registry.spid.gov.it/metadata/idp/spid-entities-idps.xml";
```

Comment:
Instead of calling external URL inside the test, use a mocked static file inside the repo provided throw git raw like [here](https://raw.githubusercontent.com/teamdigitale/io-backend/master/test_idps/spid-entities-idps.xml)